libdnf/conf/ConfigMain.cpp

```diff
@@ -593,4 +594,27 @@ OptionBool & ConfigMain::deltarpm() { return pImpl->deltarpm; }
 OptionNumber<std::uint32_t> & ConfigMain::deltarpm_percentage() { return pImpl->deltarpm_percentage; }
 OptionBool & ConfigMain::skip_if_unavailable() { return pImpl->skip_if_unavailable; }
 
+static void DIRClose(DIR *d) { closedir(d); }
+
+void ConfigMain::addVarsFromDir(std::map<std::string, std::string> & varsMap, const std::string & dirPath)
+{
+    if (DIR * dir = opendir(dirPath.c_str())) {
+        std::unique_ptr<DIR, decltype(&DIRClose)> dirGuard(dir, &DIRClose);
+        while (auto ent = readdir(dir)) {
+            auto dname = ent->d_name;
+            if (dname[0] == '.' && (dname[1] == '\0' || (dname[1] == '.' && dname[2] == '\0')))
+                continue;
+
+            auto fullPath = dirPath;
+            if (fullPath.back() != '/')
+                fullPath += "/";
+            fullPath += dname;
+            std::ifstream inStream(fullPath);
```

Comment:
Please ensure that dir is not handled.
I would also suggest that files that cannot be opened (permissions) must be skipped and reported as warning.